### PR TITLE
Added dot notation along with appropriate types

### PR DESCRIPTION
**Pull request by newclarityex**
This fork adds dot notation with working TS support and removes the 'has' function as I felt it was irrelevant.

### CODE REVIEW

**Comment by harshkhandeparkar** [comment]
Thanks a lot for your contribution!
I had one suggestion:
The `has` function was added since this library was inspired by `electron-settings` which also exports a `has` function. This library tries to make switching from electron to tauri easier by providing a similar API that can be migrated to. Could you revert the change?

I'll review the rest of the changes soon.

**Comment by newclarityex** [comment]
Oh I didn't realize, I've reverted the has change. Everything else should work mostly the same, only with added dot notation.

**Comment by harshkhandeparkar** [comment]
Thanks! I won't merge this into main yet. A few things need to be done still:
1. Updating the documentation
2. Dot notation for arrays `key1.key2[3]` which is allowed in `electron-settings`. (If it isn't possible, the documentation should be updated appropriately)

I'll create a new branch and merge this PR into it. You can contribute further if you wish or I'll handle this and try to release this ASAP!

**Comment by harshkhandeparkar** [comment]
> Dot notation for arrays key1.key2[3] which is allowed in electron-settings. (If it isn't possible, the documentation should be updated appropriately)

Is it possible?

File: src/index.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export { has, get, set } from './settings//getter-setter';
+export { get, set } from './settings//getter-setter';
```
Comment (harshkhandeparkar) [inline src/index.ts]:
```suggestion
export { has, get, set } from './settings//getter-setter';
```

File: src/settings-manager/settings-manager.ts [hunk 2]
```diff
@@ -54,20 +56,19 @@ export class SettingsManager<SettingsSchema extends {} = any> {
...
    * @param key The key for the setting
    * @returns The value of the setting
    */
-  getCache<K extends keyof SettingsSchema = keyof SettingsSchema>(key: K): SettingsSchema[K] {
-    if (this.hasCache(key)) {
-      return this.settings[key];
-    }
-    else throw 'Error: key does not exist';
+  getCache<K extends Path<SettingsSchema>>(key: K): PathValue<SettingsSchema, K> {
+    if (this.hasCache(key)) throw 'Error: key does not exist';
```
Comment (harshkhandeparkar) [inline src/settings-manager/settings-manager.ts]: Shouldn't this be `!this.hasCache`?

File: src/settings-manager/settings-manager.ts [hunk 3]
```diff
@@ -76,35 +77,26 @@ export class SettingsManager<SettingsSchema extends {} = any> {
    * @param value The new value for the setting
    * @returns The entire settings object
    */
-  setCache<K extends keyof SettingsSchema = keyof SettingsSchema>(key: K, value: SettingsSchema[K]): SettingsSchema[K] {
-    this.settings[key] = value;
+  setCache<K extends Path<SettingsSchema>, V extends PathValue<SettingsSchema, K>>(key: K, value: V): V {
+    if (this.hasCache(key)) throw 'Error: key does not exist';
```
Comment (harshkhandeparkar) [inline src/settings-manager/settings-manager.ts]: Same here?

File: src/settings/getter-setter.ts [hunk 4]
```diff
@@ -1,28 +1,31 @@
...
  */
 export async function get
-  <SettingsSchema extends {} = any, K  extends keyof SettingsSchema = keyof SettingsSchema>
-  (key: K, options: ConfigOptions = {}): Promise<SettingsSchema[K]>
-{
-  if (await has<SettingsSchema>(key)) return (await getSettings<SettingsSchema>(options)).settings[key];
-  else throw 'Error: key does not exist.'
+  <SettingsSchema, K extends Path<SettingsSchema>>
+  (key: K, options: ConfigOptions = {}): Promise<PathValue<SettingsSchema, K>> {
+  if (await has<SettingsSchema, K>(key)) throw 'Error: key does not exist';
```
Comment (harshkhandeparkar) [inline src/settings/getter-setter.ts]: And here?

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

File: src/settings-manager/settings-manager.ts (hunk 2 above)
Comment (newclarityex) [reply src/settings-manager/settings-manager.ts]: Yup

**Comment by newclarityex** [comment]
> > Dot notation for arrays key1.key2[3] which is allowed in electron-settings. (If it isn't possible, the documentation should be updated appropriately)
> 
> Is it possible?

I don't think I'm able to. electron-settings does have that functionality, but I checked and the way it's implemented has incomplete TS support.

**Comment by harshkhandeparkar** [comment]
> > > Dot notation for arrays key1.key2[3] which is allowed in electron-settings. (If it isn't possible, the documentation should be updated appropriately)
> > 
> > 
> > Is it possible?
> 
> I don't think I'm able to. electron-settings does have that functionality, but I checked and the way it's implemented has incomplete TS support.

I thought so too... This is still amazing. Thank you for your contribution!

**Pull request merged**